variable.c

```diff
@@ -134,6 +134,38 @@ rb_mod_name(VALUE mod)
     return classname(mod, &permanent);
 }
 
+// Similar to logic in rb_mod_const_get()
+static bool
+is_constand_path(VALUE name) {
+    const char *path = RSTRING_PTR(name);
+    const char *pend = RSTRING_END(name);
+
+    const char *p = path;
+
+    if (p >= pend || !*p) {
+        return false;
+    }
+
+    while (p < pend) {
+        if (p + 2 <= pend && p[0] == ':' && p[1] == ':') {
+            p += 2;
+        }
+
+        const char *pbeg = p;
+        while (p < pend && *p != ':') p++;
+
+        if (pbeg == p) return false;
+
+        VALUE part = rb_str_subseq(name, pbeg - path, p - pbeg);
+        OBJ_FREEZE(part);
+        if (!rb_is_const_name(part)) {
```

Comment:
```suggestion
        if (rb_enc_symname_type(pbeg, p - pbeg, enc, 0) != ID_CONST) {
```

Reply:
Of course `#include "symbol.h"` and a dependency in common.mk are needed, too.

Reply:
> and a dependency in common.mk are needed, too.

How to do that one?